src/meshes/t8code_fv_mesh.jl

```diff
@@ -0,0 +1,522 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    T8codeFVMesh{NDIMS} <: AbstractMesh{NDIMS}
+
+An unstructured curved mesh based on trees that uses the library `t8code`
+to manage trees and mesh refinement.
+"""
+mutable struct T8codeFVMesh{NDIMS, RealT <: Real, Forest} <: AbstractMesh{NDIMS}
```

Comment:
@bennibolm 
I think we can maintain just one `T8codeMesh` data structure which works both for `dgsem` and `fv` solvers. Once
https://github.com/trixi-framework/Trixi.jl/pull/1803 and https://github.com/trixi-framework/Trixi.jl/pull/1535 are merged into `main` we should think about how to align `T8codeMesh` with your `fv` solver implementation.